app/controllers/sessions_controller.rb

```diff
@@ -15,11 +15,26 @@ def destroy
   def failure
     respond_to do |format|
       format.html do
-        flash[:alert] = warden_message if warden_message.present?
-        redirect_to login_path
+        if request.xhr?
+          head :not_found
+        else
+          flash[:alert] = warden_message if warden_message.present?
+          session[:return_to] ||= return_to
+          redirect_to login_path
+        end
       end
       format.json { head :not_found }
       format.js { head :not_found }
     end
   end
+
+  private
+
+  def return_to
+    if request.get?
```

Comment:
Instead of adding another brakeman ignore, would it help if we follow what's recommended here: https://brakemanscanner.org/docs/warning_types/http_verb_confusion/ and add an explicit `request.post?` check?

Reply:
Gotcha!

Reply:
The warning comes up because our if block is checking `request.get?` and then we have an `else`. Brakeman is assuming the else is for `request.post?` but that's not the case in our code. This is just a false positive. If we don't want to add another ignore, then we can add an `elsif` and be explicit about what we are checking.

This fixes the warning:

```ruby
  def return_to
    if request.get?
      warden_options[:attempted_path]
    elsif request.env['HTTP_REFERER'].present?
      request.env['HTTP_REFERER']
    end
  end
```